app/src/processing/app/Base.java

```diff
@@ -925,45 +926,20 @@ public void actionPerformed(ActionEvent actionEvent) {
    */
   public boolean handleClose(Editor editor) {
     // Check if modified
-//    boolean immediate = editors.size() == 1;
     if (!editor.checkModified()) {
       return false;
     }
 
     if (editors.size() == 1) {
-      storeScreenDimensions();
-      storeSketches();
-
-      // This will store the sketch count as zero
-      editors.remove(editor);
-      try {
-        Editor.serialMonitor.close();
-      } catch (Exception e) {
-        //ignore
-      }
-      rebuildRecentSketchesMenuItems();
 
-      // Save out the current prefs state
-      PreferencesData.save();
-
-      // Since this wasn't an actual Quit event, call System.exit()
-      System.exit(0);
+      handleQuit();
```

Comment:
Isn't this something for a separate commit? Calling `handleQuit()` here seems unrelated to killing the programmer?

Reply:
Moved to own commit